ros-setup/install.bash

```diff
@@ -1,21 +1,63 @@
 #! /usr/bin/env bash
 
-if [ ! -f /etc/apt/sources.list.d/ros-latest.list ]
-then
-    tue-install-echo "Adding ROS sources to apt-get"
-    sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+keyurl="https://raw.githubusercontent.com/ros/rosdistro/master/ros.key"
+keyfile=/usr/share/keyrings/ros-archive-keyring.gpg
+
+rosrepourl="http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main"
+sourcefile=/etc/apt/sources.list.d/ros-latest.list
+sourceurl="deb [signed-by=${keyfile}] ${rosrepourl}"
 
-    tue-install-pipe sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+ADD_ROS_SOURCES=false
+ADD_ROS_GPG_KEY=false
+FORCE_UPDATE=false
 
-    tue-install-apt-get-update
-    tue-install-debug "Added ROS sources to apt-get successfully"
+if [[ ! -f "${sourcefile}" ]]
+then
+    tue-install-debug "Adding ROS sources to apt-get"
+    ADD_ROS_SOURCES=true
+    ADD_ROS_GPG_KEY=true
+elif [[ $(cat "${sourcefile}") != "${sourceurl}" ]]
```

Comment:
Better to do a grep here as there could be any comments or so in the file.

Reply:
The idea is if the files are not identical then overwrite it